tests/unit/classifier/svm/LibLinear_unittest.cc

```diff
@@ -164,20 +152,17 @@ class LibLinear : public ::testing::Test
 	}
 	void generate_data_l1()
 	{
-		initialize_train_and_test_and_ground();
-
-
-		transpose_feats_matrix();
-
-	}
-	void generate_data_l2()
-	{
-		initialize_train_and_test_and_ground();
+		generate_data_l2();
+		//Transpose the train_feats matrix
+		auto old_train_feats = train_feats;
+		train_feats = train_feats->get_transposed();
+		SG_UNREF(old_train_feats);
```

Comment:
why the unref here?

Reply:
@karlnapf  Doesn't get_transposed() create a new transposed matrix? So I'm just deallocating the old matrix (that i transposed)? Or does get_transposed() act in place?

Reply:
I guess I would just delete the old one from the same scope that you created it in (tearDown or whatever), but this is also minor.

Make sure to run the test with valgrind to see whether the memory is fine btw

Reply:
Ohh I see, get_transposed() handles the memory deallocation inside of it. So no need for SG_UNREF :)

Reply:
I dont think it does ...

Reply:
Yea i got a bit confused. Since I store the transposed matrix always in train_feats, I need to deallocate the memory before I leave the function or else we leak. The TearDown will release the transposed matrix (which is not stored in train_feats)